component/block_cache/block_cache.go

```diff
@@ -397,7 +397,7 @@ func (bc *BlockCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Han
 	log.Debug("BlockCache::OpenFile : Size of file handle.Size %v", handle.Size)
 	bc.prepareHandleForBlockCache(handle)
 
-	if options.Flags&os.O_TRUNC != 0 || (options.Flags&os.O_WRONLY != 0 && options.Flags&os.O_APPEND == 0) {
+	if options.Flags&os.O_TRUNC != 0 {
```

Comment:
Earlier we were not changing the open flags in libfuse later due to some other issues we encountered we started doing that (opening all files in RDWR mode). Hence this check is independent and not relying on what libfuse does. Tomorow if we change that logic we need to remember to correct this flag here. Also, this is just in open and one time call, so it does not have any impact on the system. I suggest we leave this here as is as some day we might correct the libfuse behavior of overriding the flags.

Reply:
I agree while write back cache is enabled then kernel also make read requests for WRONLY file mode, as we made writeback caching default we are redirecting the writeonly mode to RDWR. but the case here is for the writeonly mode when writeback cache is disabled, The default behaviour is not to change the filesize to zero.